lib/ansible/modules/remote_mount_facts.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2024 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import annotations
+
+
+DOCUMENTATION = """
+---
+module: remote_mount_facts
+version_added: 2.18
+short_description: Get mount facts for devices that do not start with a path separator.
+description:
+  - Get mount facts for devices that do not start with a path separator, which are skipped by default mount gathering.
+options:
+  collect:
+    description: A list of mount types to return.
+    type: list
+    elements: str
+    required: true
+    choices:
```

Comment:
maybe instead of an enum a filters/patterns so even if new mount types appear user can use module w/o code update?

Reply:
i.e i have vfat from my usb drive, but also hfs from a friend's mac drive i was 'rescuing', trying to keep up with the list will can be daunting

Reply:
I replaced this option with `devices` and `fstypes` to filter using fnmatch. This makes the module much easier to test too. Let me know what you think.